packages/lib/src/services/test/player-sdk.service.spec.ts

```diff
@@ -402,6 +402,63 @@ describe('Service', () => {
       });
     });
 
+    describe('ready', () => {
+      it('should execute when sdk is ready', () => {
+        expect.assertions(1);
+
+        const sdk = new PlayerSdk(iframe);
+
+        sdk.addEventListener('ready', () => {
+          expect(true).toBeTruthy();
+        });
+
+        postMessageFromPlayer<SdkReadyMessage>({
+          action: 'ready',
+          value: url.toString(),
+        });
+      });
+
+      it('should execute when handshake is done', async () => {
+        expect.assertions(2);
+
+        const callback = jest.fn().mockImplementation(() => {
+          expect(true).toBeTruthy();
+        });
+
+        const sdk = new PlayerSdk(iframe);
+
+        sdk.addEventListener('ready', callback);
+
+        postMessageFromPlayer<SdkHandshakeMessage>({
+          action: 'handshake',
+          guid,
+        });
+
+        await Promise.resolve();
+
+        expect(callback).toHaveBeenCalled();
+      });
+
+      it('should not execute when listener was removed', async () => {
+        const callback = jest.fn();
+
+        const sdk = new PlayerSdk(iframe);
+
+        sdk.addEventListener('ready', callback);
+
+        sdk.removeEventListener('ready', callback);
+
+        postMessageFromPlayer<SdkReadyMessage>({
+          action: 'ready',
+          value: url.toString(),
+        });
+
+        await Promise.resolve();
```

Comment:
why do you need this Promise.revolve()?

Reply:
since we have readyPromise and here we're in still in execution loop, Promise.resolve() will waits for readyPromise to resolve